DXRModules/DeusEx/Classes/DXRGameTimer.uc

```diff
@@ -0,0 +1,180 @@
+class DXRGameTimer extends DXRBase;
+
+var name NameConversion;
+
+function AnyEntry()
+{
+    Super.AnyEntry();
+    
+    SetTimer(0.1, True);
+}
+
+//Returns true when you aren't in a menu, or in the intro, etc.
+function bool InGame() {
+
+    if (dxr.Player.InConversation()) {
+        return True;
+    }
+
+    if (None == DeusExRootWindow(dxr.Player.rootWindow)) {
+        return False;
+    }
+    
+    if (None == DeusExRootWindow(dxr.Player.rootWindow).hud) {
+        return False;
+    }
+    
+    if (!DeusExRootWindow(dxr.Player.rootWindow).hud.isVisible()){
+        return False;
+    }
+    
+    return True;
+}
+
+function IncMissionTimer(int mission)
+{
+    local string flagname;
+    local int time;
+    
+    if (mission < 1) {
+        return;
+    }
+    
+    if (InGame()) {
+        flagname = "DXRando_Mission"$mission$"_Timer";
+        SetPropertyText("NameConversion",flagname); //Kludge for logic simplification
```

Comment:
neat trick, but you could also do `NameConversion = dxr.Player.rootWindow.StringToName(flagname);`